### DISCUSSION TITLE
Configure maxDuration for Nuxt API routes in /server directory

### DISCUSSION THREAD

**Discussion by eliabieri** [vercel/community#5257] [Help] [answered]
### Summary

I want to configure the `maxDuration` parameter for an API route in my Nuxt 3 project.
This route is called repeatedly by a Vercel cron job.

The problem is, that `vercel.json` only allows for Serverless Functions to be in the `api` folder, which is not the case for Nuxt projects. What am I missing here?

### Steps to Reproduce

1) Create a Nuxt project
2) Create an API route by creating a .ts file in the `server` directory
3) Configure the max duration for this Serverless Functions

**Comment by amyegan** [comment, marked as answer]
Great question, Elia. You an configure the `maxDuration` in your **nuxt.config.ts** file. For example:

```js
import { defineNuxtConfig } from "nuxt/config";

export default defineNuxtConfig({
  nitro: {
    vercel: {
      functions: {
        maxDuration: 60,
        memory: 1024,
      },
    },
  },
});
```

https://nuxt.com/deploy/vercel#custom-build-output-configuration
https://github.com/unjs/nitro/pull/976

**Comment by BibiSebi** [comment]
@eliabieri I am facing the same issue, and I tried the snippet from @amyegan but with not success, the maxDuration is still not applied. Did this work for you? or did you find another workaround/solution :) ?

**Reply by BibiSebi** [reply]
update: it works just took a moment to update :)

**Comment by maximilliangeorge** [comment]
Does anyone know how to do this per route? I've made various attempts.
